android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -726,6 +726,45 @@ public void peerConnectionAddICECandidate(ReadableMap candidateMap, final int id
         callback.invoke(result);
         Log.d(TAG, "peerConnectionAddICECandidate() end");
     }
+    private WritableArray convertWebRTCStats(StatsReport[] wrtcReports) {
+        WritableArray reports = Arguments.createArray();
+        for (StatsReport wrtcReport : wrtcReports) {
+            WritableMap report = Arguments.createMap();
+            report.putString("id", wrtcReport.id);
+            report.putString("type", wrtcReport.type);
+            report.putDouble("timestamp", wrtcReport.timestamp);
+            WritableArray values = Arguments.createArray();
+            for (StatsReport.Value v : wrtcReport.values) {
+                WritableMap keyValue = Arguments.createMap();
+                keyValue.putString(v.name, v.value);
+                values.pushMap(keyValue);
+            }
+            report.putArray("values", values);
+            reports.pushMap(report);
+        }
+        return reports;
+    }
+    @ReactMethod
+    public void peerConnectionGetStats(String trackId,
+                                       int id, final Callback statsCb) {
+        PeerConnection peerConnection = mPeerConnections.get(id);
+        if (peerConnection != null) {
+            MediaStreamTrack mediaTrack = mMediaStreamTracks.get(trackId);
+            if (trackId == null || trackId.isEmpty() || mediaTrack != null) {
```

Comment:
I had a dilemma on how this could be written better and your solutions looks perfect.